sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -123,26 +123,41 @@ case class ConcatWs(children: Seq[Expression])
       val idxVararg = ctx.freshName("idxInVararg")
 
       val evals = children.map(_.genCode(ctx))
+      val supportSplitExpressions = ctx.supportSplitExpressionsWithCurrentInputs
+
+      // Here we apply different approaches according to the availability of splitting expressions.
```

Comment:
It would be good to use one approach to reduce the complexity. Could you please see @cloud-fan 's comment?

Reply:
I'll go through the comment. Thanks.